### PR TITLE
fix(idd-ci): document --watch --required non-block case

### LINKED ISSUE #2796: fix(idd-ci): document that --watch --required does not block with no required checks
## Background

`.github/instructions/idd-ci.instructions.md`'s Wake-up discipline section
presents `gh pr checks <pr-number> --watch --required` as the canonical
synchronous blocking wait for CI, with no caveat. The same file's separate
"Required-check discovery" section already names and handles
`noRequiredChecksConfigured` as a real, intentional state -- but only as an
*interpretation-time* concern (how to read an empty required-check set),
never as a *wait-mechanism-selection* concern.

Confirmed live: `gh pr checks <n> --watch --required` returns
**immediately**, non-blocking, printing a "no required checks reported on
the '<branch>' branch" message, on a repository confirmed to have zero
required checks configured for the PR's base branch -- even while the PR's
own real CI checks were still running. An agent following the Wake-up
discipline text literally, without independently cross-referencing the
Required-check discovery section first, can read this instant return as
"nothing to wait for, proceed" and move past an unfinished CI run.

`idd-skill#1807` (closed) is a related but distinct issue: it addresses the
blocking watch call's own execution-timeout ceiling, not this non-blocking
early-return case, which happens regardless of any timeout.

## Proposed change

Add an explicit caveat to the Wake-up discipline section's exemplar
command: when the repository has no required checks configured for the
target branch (the same `noRequiredChecksConfigured` state the Required-check
discovery section already names), `gh pr checks --watch --required` is not
a blocking wait at all, and the caller must instead block on the bare
`gh pr checks <pr-number> --watch` (no `--required`) or another mechanism
that does not depend on a required-check rollup existing. Cross-reference
the Required-check discovery section from Wake-up discipline so a caller
resolves `noRequiredChecksConfigured` before choosing which watch command
to issue.

## Acceptance criteria

- Wake-up discipline's blocking-wait guidance states explicitly that
  `--watch --required` does not block when
  `noRequiredChecksConfigured` is true, and names the alternative
  (the bare `--watch` form or an equivalent non-required-rollup-dependent
  wait).
- The section cross-references "Required-check discovery" so a caller
  resolving this state before choosing a wait mechanism is the documented
  path, not an implicit inference.
- `node scripts/audit-docs.mjs --check` (or the profile-equivalent) still
  passes after the edit.

## Candidate files

- `.github/instructions/idd-ci.instructions.md`
- `idd-template/.github/instructions/idd-ci.instructions.md`

---

_Autopilot suitability: 4 / 5 -- higher is more autopilot-suitable;
below the configured floor is human-oriented._

_Effort: S -- author-estimated size; a soft autopilot selection
tie-breaker only._

### PR DESCRIPTION
**Pull request by kurone-kito**
## Summary

Wake-up discipline presented `gh pr checks --watch --required` as
the canonical blocking wait for CI with no caveat, but the same
file's Required-check discovery section already treats
`noRequiredChecksConfigured` as a real, intentional state. Confirmed
live: `--watch --required` returns immediately, non-blocking,
printing "no required checks reported" even while real CI is still
running, on a repository with no required checks configured for the
PR's base branch. A session following Wake-up discipline literally,
without cross-referencing Required-check discovery first, could read
the instant return as "nothing to wait for" and move past unfinished
CI.

Adds the caveat, names the fallback (bare `--watch`, no
`--required`), and cross-references Required-check discovery so
resolving `noRequiredChecksConfigured` before choosing a wait command
is the documented path.

## Verification

- All 3 acceptance criteria from the issue satisfied
- `node scripts/sync-docs.mjs --apply` (mirror in sync)
- `node scripts/audit-docs.mjs --check` (pass; `audit/sync-manifest.json`
  untouched)
- `pnpm run typecheck`, `pnpm test` (5524/5524 pass), markdownlint,
  dprint, cspell, `audit-code-span-wrap.mjs` all clean
- Report-only critique fork found no blocking issues; applied all 3
  of its clarity/precision/style suggestions (disambiguate "Both"/
  "Neither" now that a third wait command is named, match the exact
  "no required checks reported" CLI message text, un-bold the caveat
  body to match the file's existing convention)

Closes #2796

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01DDJpTELrY2SnjHW9m5Aw7i

## Summary by CodeRabbit

- **Documentation**
  - Updated CI monitoring guidance to use required-check monitoring when required checks are configured.
  - When no required checks are configured, guidance now uses general CI monitoring so running checks remain visible.
  - Clarified that CI watch commands do not monitor Copilot review status.

### CODE REVIEW

File: .github/instructions/idd-ci.instructions.md [hunk 1]
```diff
@@ -340,9 +340,16 @@ condition below accounts for this.
   completion, or background only if the topology is confirmed to route
   completion back to this turn; otherwise wait synchronously — block
   with `gh pr checks <pr-number> --watch --required` (works on a
-  fine-grained PAT; `gh run watch <run-id> --exit-status` does not).
-  Both only block, never decide: required-only scoping, duplicate-name
-  collapse, the no-required-checks route, and the
+  fine-grained PAT; `gh run watch <run-id> --exit-status` does not) —
+  **but only once** [Required-check discovery](#required-check-discovery)
+  has resolved `noRequiredChecksConfigured: false`. When it resolved
+  `true`, `--watch --required` returns immediately, non-blocking,
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply .github/instructions/idd-ci.instructions.md]: **Accepted** — fixed in `517469d57d44d7dcda9cf7d8a2c8cf412770bee7`: named the field explicitly ('When `noRequiredChecksConfigured` resolved `true`') instead of 'it'.

File: idd-template/.github/instructions/idd-ci.instructions.md [hunk 2]
```diff
@@ -335,9 +335,16 @@ condition below accounts for this.
   completion, or background only if the topology is confirmed to route
   completion back to this turn; otherwise wait synchronously — block
   with `gh pr checks <pr-number> --watch --required` (works on a
-  fine-grained PAT; `gh run watch <run-id> --exit-status` does not).
-  Both only block, never decide: required-only scoping, duplicate-name
-  collapse, the no-required-checks route, and the
+  fine-grained PAT; `gh run watch <run-id> --exit-status` does not) —
+  **but only once** [Required-check discovery](#required-check-discovery)
+  has resolved `noRequiredChecksConfigured: false`. When it resolved
+  `true`, `--watch --required` returns immediately, non-blocking,
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-ci.instructions.md]: **Accepted** — fixed in `517469d57d44d7dcda9cf7d8a2c8cf412770bee7`: same fix as the mirror copy — named the field explicitly instead of 'it'.

File: idd-template/.github/instructions/idd-ci.instructions.md [hunk 3]
```diff
@@ -335,9 +335,16 @@ condition below accounts for this.
...
+  printing a "no required checks reported" message even while real CI
+  is still running — block with the bare
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-ci.instructions.md]: **Rejected** — this is already covered by an explicit policy exemption, not an omission. `docs/idd-design-rationale.md` "Cite the observed incident" (around line 847-856) states: "That exemption is not retrofit-only — a brand-new `.github/instructions/` passage that names an anti-pattern or failure mode is exempt from the citation requirement too, for the same reason: those bundle budgets already sit near their ceiling." This new passage is exactly that case: a brand-new `.github/instructions/` passage naming a failure mode, so no citation is required. AGENTS.md's summary bullet (which this review cites) doesn't carry that carve-out; the fuller rationale doc does. Refs #2796

**Comment by kurone-kito** [comment]
advisory-wait: claude-idd2-1506c0ce 517469d57d44d7dcda9cf7d8a2c8cf412770bee7 2026-09-09T14:41:15Z

File: .github/instructions/idd-ci.instructions.md [hunk 4]
```diff
@@ -340,9 +340,17 @@ condition below accounts for this.
   completion, or background only if the topology is confirmed to route
   completion back to this turn; otherwise wait synchronously — block
   with `gh pr checks <pr-number> --watch --required` (works on a
-  fine-grained PAT; `gh run watch <run-id> --exit-status` does not).
-  Both only block, never decide: required-only scoping, duplicate-name
-  collapse, the no-required-checks route, and the
+  fine-grained PAT; `gh run watch <run-id> --exit-status` does not) —
+  **but only once** [Required-check discovery](#required-check-discovery)
+  has resolved `noRequiredChecksConfigured: false`. When
+  `noRequiredChecksConfigured` resolved `true`, `--watch --required`
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply .github/instructions/idd-ci.instructions.md]: **Accepted** — fixed in `64fcb8626badd9580e34813443025d9f7e2f3a7a`: both branches now use perfect tense ('has resolved `noRequiredChecksConfigured: false`' / 'has resolved `noRequiredChecksConfigured: true`').

File: idd-template/.github/instructions/idd-ci.instructions.md [hunk 5]
```diff
@@ -335,9 +335,17 @@ condition below accounts for this.
   completion, or background only if the topology is confirmed to route
   completion back to this turn; otherwise wait synchronously — block
   with `gh pr checks <pr-number> --watch --required` (works on a
-  fine-grained PAT; `gh run watch <run-id> --exit-status` does not).
-  Both only block, never decide: required-only scoping, duplicate-name
-  collapse, the no-required-checks route, and the
+  fine-grained PAT; `gh run watch <run-id> --exit-status` does not) —
+  **but only once** [Required-check discovery](#required-check-discovery)
+  has resolved `noRequiredChecksConfigured: false`. When
+  `noRequiredChecksConfigured` resolved `true`, `--watch --required`
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-ci.instructions.md]: **Accepted** — fixed in `64fcb8626badd9580e34813443025d9f7e2f3a7a`: same fix as the mirror copy — both branches now use perfect tense.

File: idd-template/.github/instructions/idd-ci.instructions.md [hunk 6]
```diff
@@ -335,9 +335,17 @@ condition below accounts for this.
...
+  returns immediately, non-blocking, printing a "no required checks
+  reported" message even while real CI is still running — block with
+  the bare `gh pr checks <pr-number> --watch` (no `--required`)
+  instead. Both
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-ci.instructions.md]: **Accepted** — fixed in `64fcb8626badd9580e34813443025d9f7e2f3a7a`: verified against tests/required-checks-summary.test.mts's #2714 lone-CANCELLED-run test, which confirmed the edge case is real. Named it explicitly right where the bare --watch fallback is introduced, and reinforced that all three watch forms only block, never decide -- the actual verdict always comes from the algorithm, not from any watch's return alone.

File: .github/instructions/idd-ci.instructions.md [hunk 7]
```diff
@@ -340,9 +340,19 @@ condition below accounts for this.
   completion, or background only if the topology is confirmed to route
   completion back to this turn; otherwise wait synchronously — block
   with `gh pr checks <pr-number> --watch --required` (works on a
-  fine-grained PAT; `gh run watch <run-id> --exit-status` does not).
-  Both only block, never decide: required-only scoping, duplicate-name
-  collapse, the no-required-checks route, and the
+  fine-grained PAT; `gh run watch <run-id> --exit-status` does not) —
+  **but only once** [Required-check discovery](#required-check-discovery)
+  has resolved `noRequiredChecksConfigured: false`. When it has
+  resolved `noRequiredChecksConfigured: true`, `--watch --required`
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply .github/instructions/idd-ci.instructions.md]: **Accepted** — fixed in `97b481af4a5e1d93a830ffcfae3698ced7407920`: named 'Required-check discovery' explicitly for the true branch too, removing the reintroduced pronoun.

File: idd-template/.github/instructions/idd-ci.instructions.md [hunk 8]
```diff
@@ -335,9 +335,19 @@ condition below accounts for this.
   completion, or background only if the topology is confirmed to route
   completion back to this turn; otherwise wait synchronously — block
   with `gh pr checks <pr-number> --watch --required` (works on a
-  fine-grained PAT; `gh run watch <run-id> --exit-status` does not).
-  Both only block, never decide: required-only scoping, duplicate-name
-  collapse, the no-required-checks route, and the
+  fine-grained PAT; `gh run watch <run-id> --exit-status` does not) —
+  **but only once** [Required-check discovery](#required-check-discovery)
+  has resolved `noRequiredChecksConfigured: false`. When it has
+  resolved `noRequiredChecksConfigured: true`, `--watch --required`
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-ci.instructions.md]: **Accepted** — fixed in `97b481af4a5e1d93a830ffcfae3698ced7407920`: same fix as the mirror copy — named 'Required-check discovery' explicitly, removing the pronoun.

**Comment by kurone-kito** [comment]
advisory-wait: claude-idd2-1506c0ce 97b481af4a5e1d93a830ffcfae3698ced7407920 2026-09-09T14:59:03Z

File: .github/instructions/idd-ci.instructions.md [hunk 9]
```diff
@@ -340,9 +340,20 @@ condition below accounts for this.
...
+  reported" message even while real CI is still running — block with
+  the bare `gh pr checks <pr-number> --watch` (no `--required`)
+  instead. That bare form still returns once every visible check
+  reaches a terminal GitHub state, which is not the same as "safe to
+  proceed" — a lone `CANCELLED` check with no same-producer successor
+  is one such terminal-but-`pending` case (#2714). All three watch
+  forms above only block, never decide: required-only scoping,
+  duplicate-name collapse, the no-required-checks route, and the
   `ciWait.runningTimeout`/`generationTimeout` bound all stay with the
   algorithm above — track elapsed time and apply its rerun-or-hold
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply .github/instructions/idd-ci.instructions.md]: **Accepted** — fixed in `fc5287895fa4bceb82a2e489cfce3b065f9697d4`: reworded to 'None of the three watch forms above decides anything by itself ... these are all interpretation-time concerns that stay with the algorithm above regardless of which form blocked the wait' -- makes clear these are algorithm-side concerns, not per-command properties.

File: idd-template/.github/instructions/idd-ci.instructions.md [hunk 10]
```diff
@@ -335,9 +335,20 @@ condition below accounts for this.
...
+  reported" message even while real CI is still running — block with
+  the bare `gh pr checks <pr-number> --watch` (no `--required`)
+  instead. That bare form still returns once every visible check
+  reaches a terminal GitHub state, which is not the same as "safe to
+  proceed" — a lone `CANCELLED` check with no same-producer successor
+  is one such terminal-but-`pending` case (#2714). All three watch
+  forms above only block, never decide: required-only scoping,
+  duplicate-name collapse, the no-required-checks route, and the
   `ciWait.runningTimeout`/`generationTimeout` bound all stay with the
   algorithm above — track elapsed time and apply its rerun-or-hold
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-ci.instructions.md]: **Accepted** — fixed in `fc5287895fa4bceb82a2e489cfce3b065f9697d4`: same fix as the mirror copy.

File: idd-template/.github/instructions/idd-ci.instructions.md [hunk 11]
```diff
@@ -335,9 +335,20 @@ condition below accounts for this.
...
-  Both only block, never decide: required-only scoping, duplicate-name
-  collapse, the no-required-checks route, and the
+  fine-grained PAT; `gh run watch <run-id> --exit-status` does not) —
+  **but only once** [Required-check discovery](#required-check-discovery)
+  has resolved `noRequiredChecksConfigured: false`. When Required-check
+  discovery has instead resolved `noRequiredChecksConfigured: true`,
+  `--watch --required` returns immediately, non-blocking, printing a
+  "no required checks
+  reported" message even while real CI is still running — block with
+  the bare `gh pr checks <pr-number> --watch` (no `--required`)
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-ci.instructions.md:345]: **Rejected** — already correctly handled, not a gap. `ciWait.generationTimeout` (line ~302 of this same file: "Required checks are not generated after `ciWait.generationTimeout` | Treat as running (default 10 min)...") is the established mechanism for exactly this zero-visible-checks-immediately-after-push state -- the algorithm treats it as 'running', not 'hold', for up to that 10-minute window, so normal check-generation latency does not produce a false hold. The sentence right before this one already tells the reader that the no-required-checks route and the generationTimeout bound both 'stay with the algorithm above' rather than being decided by any watch command, which is exactly the deferral this finding is asking for. Refs #2796

**Comment by kurone-kito** [comment]
advisory-wait: claude-idd2-1506c0ce fc5287895fa4bceb82a2e489cfce3b065f9697d4 2026-09-09T15:09:47Z

**Comment by kurone-kito** [comment]
**Rejected** — coderabbitai[bot] did not review HEAD fc5287895fa4bceb82a2e489cfce3b065f9697d4 (review limit reached / rate limited); this is not a completed review (source: #issuecomment-5603564097)

**Comment by kurone-kito** [comment]
**Accepted** — chatgpt-codex-connector[bot] summary walkthrough at HEAD fc5287895fa4bceb82a2e489cfce3b065f9697d4; actionable comments, if any, are dispositioned as their own review threads

**Comment by kurone-kito** [comment]
_claude-idd2-1506c0ce: review triage snapshot — IDD automation marker. Do not edit._
